Core/Foundation/Source/DebugTrace.cpp

```diff
@@ -0,0 +1,45 @@
+#include <stdarg.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <functional>
+
+namespace Babylon
+{
+    namespace
+    {
+        static bool s_debugTraceEnabled{};
+        static std::function<void(const char* output)> s_traceOutputFunction;
+        static void TraceArgs(const char* format, va_list args)
+        {
+            char temp[8192];
+            auto len = vsnprintf(temp, sizeof(temp), format, args);
+            temp[len] = '\0';
```

Comment:
`vsnprintf` may return negative values if an error occurs. Maybe should check that.

Reply:
Fixed.